rubber_duck.py

```diff
@@ -1,156 +1,184 @@
+
+    async def __call__(self, thread_id: int, engine: str, prompt: str, initial_message: Message, timeout=600):
+        return await self.have_conversation(thread_id, engine, prompt, initial_message, timeout)
+
+    #
+    # Begin Conversation
+    #
+    async def have_conversation(self, thread_id: int, engine: str, prompt: str, initial_message: Message, timeout=600):
+        async with queue('messages', str(thread_id)) as messages:
+            message_history = [
+                GPTMessage(role='system', content=prompt)
+            ]
+            user_id = initial_message['author_id']
+            guild_id = initial_message['guild_id']
+            await self._metrics_handler.record_message(
+                guild_id, thread_id, user_id, message_history[0]['role'], message_history[0]['content'])
+
+            await self._send_message(thread_id, f'Hello {initial_message["author_mention"]}, how can I help you?')
+
+            while True:
+                # TODO - if the conversation is getting long, and the user changes the subject
+                #  prompt them to start a new conversation (and close this one)
+                try:
+                    message: Message = await asyncio.wait_for(messages.get(), timeout)
+
+                except asyncio.TimeoutError:
+                    await self._send_message(thread_id, '*This conversation has been closed.*')
+                    return
+
+                if len(message['file']) > 0:
+                    await self._send_message(thread_id, "I'm sorry, I can't read file attachments. Please resend your message with the relevant parts of your file included in the message.")
+                    continue
+
+                message_history.append(GPTMessage(role='user', content=message['content']))
+
+                # replaced by recommended lines (102-104) above line 106 (101 originally)
+                # if message['is_file'] == True:
+                #     appended_content = "*The user attached a file*"
+                # else:
+                #     appended_content = message['content']
+
+                # message_history.append(GPTMessage(role='user', content=appended_content))
+
+                user_id = message['author_id']
+                guild_id = message['guild_id']
+
+                await self._metrics_handler.record_message(
+                    guild_id, thread_id, user_id, message_history[-1]['role'], message_history[-1]['content'])
+
+                try:
+                    if message_history[-1]['content'] == "*The user attached a file*":
+                        response = "I'm unable to read entire files directly. Could you please paste the relevant sections and rephrase your question? " \
+                                   "This will help me assist you better!"
+
+                        await self._metrics_handler.record_message(
+                            guild_id, thread_id, user_id, 'assistant', response)

```

Comment:
What is this for?

It looks extremely brittle. Comparing the last message content to a hard-coded string is bound to break.

Reply:
It looks like a leftover from the code commented out on line 115. I think we should be able to delete this entire `if` block and just have the `else` logic.